### DISCUSSION TITLE
Option to disable folding

### DISCUSSION THREAD

**Discussion by Irmitya** [microsoft/pylance-release#5719] [Enhancement]
In #3877  / V2022.1.1, Pylance heavily changed the folding results. 
In order to get back regular folding, you have to either 
1. rollback to V2022.1.0 (or earlier) blocking access to whatever new features are added afterwards
2. disable Pylance, removing the Rename Symbol option (broken without Pylance)

I'm requesting an [option to Disable Pylance's version of Folding].
So that I can update Pylance without breaking folding for me.

**Comment by Irmitya** [comment]
VSCode has a setting for folding, that allows custom folding for languages, or the default based on indentation.
It's a one-off setting, applied to all languages, 
The 2022.1.1 update added custom "LSP" folding to python. 

I don't know what any other language's custom folding looks like but here, it's just culling code from the folding hierarchy and essentially just overfolding. 
Disabling the folding option get's back the regular folding, without the need to downgrade/remove.
It then disables custom folding for other languages but again, I don't know what it's "meant" to do here or what any other would do.

**Reply by heejaechang** [reply]
is making option to apply specific to python also doesn't work?

```
"[python]": {
   . ...
}
```

**Reply by Irmitya** [reply]
Thank you, that works.  
I had settings like that in my file but I forget what they did, so I didn't realize that was a possibility.

I can also now see that pylance isn't the only one doing this folding; json is folding like that too. I'm starting to think this folding thing is just meant to specify what is a comment in the language, to have the folding ignore those lines 🤔.
